frontend/src/scenes/session-recordings/player/playlist-popover/PlaylistPopover.tsx

```diff
@@ -95,9 +101,16 @@ export function PlaylistPopoverButton(props: LemonButtonProps): JSX.Element {
                                                     <LemonCheckbox className="pointer-events-none" checked={selected} />
                                                 )
                                             }
-                                            onClick={() =>
+                                            onClick={() => {
+                                                if (
+                                                    setPinnedInCurrentPlaylist &&
+                                                    playlist.short_id === currentPlaylistId
+                                                ) {
+                                                    return setPinnedInCurrentPlaylist(!selected)
+                                                }
+
```

Comment:
So initially it seemed like this should go into `playlistPopoverLogic`: extend it with the currently shown playlist id, and then conditionally call the endpoint or not.

However that would not update the pinned state in `sessionRecordingsPlaylistLogic`. Is there a way for that logic to be aware of an action being called in another logic, and react to that?

Reply:
> Is there a way for that logic to be aware of an action being called in another logic, and react to that?

there is!

will dig out an example

Reply:
example where you don't need props for the logic https://github.com/PostHog/posthog/blob/c5dc3b7067a5bfb70cea54d40ec65e51239d7300/frontend/src/lib/components/AddToDashboard/addToDashboardModalLogic.ts#L94